lib/gemini.js

```diff
@@ -23,39 +26,49 @@ function requireWithNoCache(moduleName) {
     return result;
 }
 
-function Gemini(config, overrides) {
-    config = config || DEFAULT_CFG_NAME;
-    this.config = new Config(config, overrides);
-    var _this = this;
+module.exports = inherit(EventEmitter, {
+    __constructor: function(config, overrides) {
+      config = config || DEFAULT_CFG_NAME;
```

Comment:
>At some point I was considering adding calls before and after calling the gemini.test but dropped the idea afterwards because then it wouldn't work when running tests programmatically using the API, right?

Yes, you are totally right, if we are adding plugins they should be loaded when using `gemini` via API, CLI and GUI